exercises/wordy/src/example/java/WordProblemSolver.java

```diff
@@ -0,0 +1,60 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+final class WordProblemSolver {
+
+    private static final String INTEGER_REGEX_STRING = "(-?\\d+)";
+
+    private static final String BINARY_OPERATION_REGEX_STRING
+        = "(plus|minus|multiplied by|divided by|raised to the power)";
+
+    private static final String VALID_QUESTION_REGEX_STRING =
+            "^What is " + INTEGER_REGEX_STRING +
+            "( " + BINARY_OPERATION_REGEX_STRING + " " + INTEGER_REGEX_STRING + ")+" +
+            "\\?$";
+
+    int solve(final String wordProblem) {
+        if (!wordProblem.matches(VALID_QUESTION_REGEX_STRING)) {
+            throw new IllegalArgumentException("I'm sorry, I don't understand the question!");
+        }
+
+        final Matcher initialValueMatcher = Pattern.compile(INTEGER_REGEX_STRING).matcher(wordProblem);
+
+        initialValueMatcher.find();
+        int result = Integer.parseInt(initialValueMatcher.group());
+
+        final Matcher operationAndValueMatcher
+                = Pattern.compile(BINARY_OPERATION_REGEX_STRING + " " + INTEGER_REGEX_STRING).matcher(wordProblem);
+
+        while (operationAndValueMatcher.find()) {
+            result = applyBinaryOperation(
+                result,
+                operationAndValueMatcher.group(1),
+                Integer.parseInt(operationAndValueMatcher.group(2)));
+        }
+
+        return result;
+    }
+
+    private int applyBinaryOperation(
+            final int firstValue,
+            final String operationDescription,
+            final int secondValue) {
+
+        switch (operationDescription) {
+            case "plus":
+                return firstValue + secondValue;
+            case "minus":
+                return firstValue - secondValue;
+            case "multiplied by":
+                return firstValue * secondValue;
+            case "divided by":
+                return firstValue / secondValue;
+            case "raised to the power":
```

Comment:
I'll blame the cold. 😬 I now understand your original question.

It's not obvious to me...

- it's a bonus iteration, so this would indicate to me a requirement that likely causes the practitioner to rethink their whole solution.
  - In general, I wish we had something like this for all exercises to both encourage exploration and increase fluency in the skill of deferring design choices that can be deferred (i.e. make the solution as concrete as feasible and generalize as requirements arrive).
  - even better is a pair of conflicting bonuses
- the practitioner is being asked to write a parser, if the requirement encourages them to "mature" that part of the code, seems on target.
- however, it's kindof a one-off... while that is a nice simulation of real-world software, it's a bit frustrating to the Exercism practitioner.

In the sum, I personally come down on the side of supporting the additional sentence structure.

_That said_, it is truly up to you: if you dig writing the more general solution, go for it.  If you either disagree with the sum or you agree but want to get this off your plate,  we can take this for now and amend the exercise later (if someone wants to).  There's no additional cost to deferring.